src/main/java/at/bitfire/jtx/JtxContract.kt

```diff
@@ -0,0 +1,1081 @@
+         * of the recurrence instance.
+         */
+        const val RECURID = "recurid"
+
+        /**
+         * Stores the reference to the original event from which the recurring event was derived.
+         * This value is NULL for the orignal event.
+         * Type: [Long]
+         */
+        const val RECUR_ORIGINALICALOBJECTID = "original_id"
+
+
+        /**
+         * Purpose:  This property defines the persistent, globally unique identifier for the calendar component.
+         * See [https://tools.ietf.org/html/rfc5545#section-3.8.4.7]
+         * Type: [String]
+         */
+        const val UID = "uid"
+
+        /**
+         * Purpose:  This property specifies the date and time that the calendar information
+         * was created by the calendar user agent in the calendar store.
+         * See [https://tools.ietf.org/html/rfc5545#section-3.8.7.1]
+         * Type: [Long]
+         */
+        const val CREATED = "created"
+
+        /**
+         * Purpose:  In the case of an iCalendar object that specifies a
+         * "METHOD" property, this property specifies the date and time that
+         * the instance of the iCalendar object was created.  In the case of
+         * an iCalendar object that doesn't specify a "METHOD" property, this
+         * property specifies the date and time that the information
+         * associated with the calendar component was last revised in the
+         * calendar store.
+         * See [https://tools.ietf.org/html/rfc5545#section-3.8.7.2]
+         * Type: [Long]
+         */
+        const val DTSTAMP = "dtstamp"
+
+        /**
+         * Purpose:  This property specifies the date and time that the information associated
+         * with the calendar component was last revised in the calendar store.
+         * See [https://tools.ietf.org/html/rfc5545#section-3.8.7.3]
+         * Type: [Long]
+         */
+        const val LAST_MODIFIED = "lastmodified"
+
+        /**
+         * Purpose:  This property defines the revision sequence number of the calendar component within a sequence of revisions.
+         * See [https://tools.ietf.org/html/rfc5545#section-3.8.7.4]
+         * Type: [Int]
+         */
+        const val SEQUENCE = "sequence"
+
+        /**
+         * Purpose:  This property specifies a color used for displaying the calendar, event, todo, or journal data.
+         * See [https://tools.ietf.org/html/rfc7986#section-5.9]
+         * The expected String is a String that can be parsed by Color.parseColor(...)
+         * Type: [String]
+         */
+        const val COLOR = "color"
+
+        /**
+         * Purpose:  This column is the foreign key to the [JtxCollection].
+         * Type: [Long]
+         */
+        const val ICALOBJECT_COLLECTIONID = "collectionId"
+
+        /**
+         * Purpose:  This column defines if a local collection was changed that is supposed to be synchronised.
```

Comment:
collection und/oder object? Wird das wirklich nur bei zu synchronisierenden Collections gesetzt? Falls man über den Contract eh nur auf diese zugreifen kann, kann man es weglassen.

Reply:
Ah, das war ein Fehler, das ist nur für das ICalObject. Die Collections können in jtx Board nicht verändert werden, das Dirty flag gibt's eh nicht für Collections.